### DISCUSSION TITLE
Not able to bundle react component with zustand when using Rollup

### DISCUSSION THREAD

**Discussion by tillganster** [pmndrs/zustand#2262] [Bug report]
I have a react component which is bundled using rollup. Currently I am trying to integrate zustand into this component. Unfortunatly I am getting the following error:

```bash 
[!] RollupError: Could not resolve "./react.mjs" from "node_modules/zustand/esm/index.d.mts"
```

Having a look into `node_modules/zustand/esm` shows that there is no `react.mjs`. Is this a problem of my rolllup configuration or a bug in zustand ?

**Comment by dbritto-dev** [comment]
@tillganster weird, should be there

**Reply by tillganster** [reply]
![Bildschirmfoto 2023-12-19 um 16 50 49](https://github.com/pmndrs/zustand/assets/49649012/7368d38c-a8fa-47e6-9b32-12e64c3fcb40)
Seen here in code Tab on npm, which is the same in my local machine: https://www.npmjs.com/package/zustand/v/4.4.7?activeTab=code

**Reply by dai-shi** [reply]
https://www.typescriptlang.org/docs/handbook/modules/reference.html#file-extension-substitution
<img width="531" alt="image" src="https://github.com/pmndrs/zustand/assets/490574/d4147ad1-1b49-4c0e-b01a-5b6e3ca41327">

We expect it to look for `react.d.mts`. I'm not 100% happy with the current output. I hope `tsc` has an option to emit with changing the extension, or even bundle TS files.

That said, such a rollup error hasn't been reported AFAIR. Is it particularly a new version of rollup?
